### PR TITLE
docs(OpenAPI): enhance OpenAPI descriptions for endpoints, request an…

### PR DESCRIPTION
**Pull request by nicoprow**
SpringDoc OpenAPI can't easily have custom schema names for generic types. In order to provide custom names the effort currently doesn't justify the benefit. That's why you will run across some strange names for some schemas there.

### CODE REVIEW

File: src/main/kotlin/com/catenax/gpdm/dto/request/BusinessPartnerSearchRequest.kt [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
+    @Parameter(description = "Which business partner name")
     val name: String?,
+    @Parameter(description = "Which legal form name")
     val legalForm: String?,
+    @Parameter(description = "Which business status designation")
     val status: String?,
+    @ParameterObject
     var address: AddressSearchRequest?,
+    @Parameter(description = "Which business classification")
     val classification: String?,
```
Comment (pazepaze) [inline src/main/kotlin/com/catenax/gpdm/dto/request/BusinessPartnerSearchRequest.kt:19]: Swagger Annotation doesn't seem to work.
I think ```@field:Parameter(description = ...)``` needs to be used here.

**Review by pazepaze — CHANGES_REQUESTED** [review]
the rest looks good

File: src/main/kotlin/com/catenax/gpdm/dto/request/BusinessPartnerSearchRequest.kt (hunk 1 above)
Comment (nicoprow) [reply src/main/kotlin/com/catenax/gpdm/dto/request/BusinessPartnerSearchRequest.kt]: Right, with the @field they would be visible, but actually those classes will be covered in pull request #40 as well. So I just deleted the annotations for now as those classes have additional problems: @ParameterObject is not working here for example as well. Fixes for that are being taken care of in #40 .

**Review by pazepaze — COMMENTED** [review]
As we discussed in the daily, you wanted to think about the sonar motivated changes

**Review by pazepaze — APPROVED** [review]

**Comment by nicoprow** [comment]
Removed OpenApi descriptions from the values again as discussed. But reduced a bit of duplication by introducing a constant holding identical descriptions.

**Pull request merged**